go/vt/sqlparser/sql.y

```diff
@@ -2836,6 +2836,10 @@ set_expression:
   {
     $$ = &SetExpr{Name: NewColIdent(string($1)), Expr: $2}
   }
+| charset_or_character_set charset_value COLLATE charset_value
```

Comment:
Instead of another rule here, it'd be better to change the definition of `collate_opt` to use `charset_value` instead of just `ID`.